### ISSUE TITLE
Add installation of the binary to build.sh

### ISSUE THREAD

**Issue by aleksijohansson** [wunderkraut/radi-cli#7]
Currently the build.sh script only builds the binary, but does not install it. It would be good addition to install it. On macOS we can use `/usr/local/bin/` folder which is properly configured already if Homebrew is in use.

**Comment by james-nesbitt** [comment]
are we sure that OSX will give users permission to install to that location?  I thought that recent versions of OSX prevented user writing to that path?

**Comment by james-nesbitt** [comment]
I will add this option as directed, but I will wrap it in a bash question/confirmation.

**Comment by james-nesbitt** [comment]
so far I have a decent approach: 

```
-> % ./build.sh
***** Building RADI cli client.

This will build the radi-cli as a 'radi' binary for 'linux-amd64'. 

(Override this by setting $GOOS and $GOARCH environment variables)

 **** Building in containerized golang environment
 
"./build/make.sh" getdeps
---> Make-bundle: getdeps (in .)
  --> updating go dependencies
     - golang.org/x/net/context
     - github.com/Sirupsen/logrus
     - gopkg.in/urfave/cli.v2
     - gopkg.in/yaml.v2
     - github.com/docker/libcompose
     - github.com/Jalle19/upcloud-go-sdk/upcloud/
     - github.com/Jalle19/upcloud-go-sdk/upcloud/client
     - github.com/rancher/go-rancher/v2
     - github.com/james-nesbitt/init-go
     - github.com/wunderkraut/radi-api
     - github.com/wunderkraut/radi-handlers
  --- Finished dependency update

"./build/make.sh" binary
---> Make-bundle: binary (in .)
  --> Building ./bin/radi

 

Exited container

 **** Containerized build complete 

an executable binary has (hopefully) now been built 
in ./bin/radi


 **** Installation

This installer can now install the built binary for you,
if you don't want to do it manually.

The planned installation path is : /usr/local/bin

Would you like to me install a binary to that location? (y/n)

y
"./build/make.sh" install
---> Make-bundle: install (in .)
  --> Installing last compiled binary (./bin/radi) to: /usr/local/bin
cp: cannot create regular file '/usr/local/bin/radi': Permission denied
make: *** [Makefile:19: install] Error 1
```

I have left the last step as requiring a sudo, which I will put in later.

**Comment by james-nesbitt** [comment]
@aleksijohansson please review https://github.com/wunderkraut/radi-cli/issues/14

**Comment by aleksijohansson** [comment]
@james-nesbitt Please add the permission escalation part and then I can review this.

**Comment by james-nesbitt** [comment]
latest changes in that branch now produce this output:

```
-> % ./build.sh
***** Building RADI cli client.

This will build the radi-cli as a 'radi' binary for 'linux-amd64'. 

(Override this by setting $GOOS and $GOARCH environment variables)

 **** Building in containerized golang environment
 
"./build/make.sh" clean
---> Make-bundle: clean (in .)
  --> Cleaning by removing build : ./bin/radi

"./build/make.sh" getdeps
---> Make-bundle: getdeps (in .)
  --> updating go dependencies
     - golang.org/x/net/context
     - github.com/Sirupsen/logrus
     - gopkg.in/urfave/cli.v2
     - gopkg.in/yaml.v2
     - github.com/docker/libcompose
     - github.com/Jalle19/upcloud-go-sdk/upcloud/
     - github.com/Jalle19/upcloud-go-sdk/upcloud/client
     - github.com/rancher/go-rancher/v2
     - github.com/james-nesbitt/init-go
     - github.com/wunderkraut/radi-api
     - github.com/wunderkraut/radi-handlers
  --- Finished dependency update

"./build/make.sh" binary
---> Make-bundle: binary (in .)
  --> Building ./bin/radi
github.com/wunderkraut/radi-api/operation
github.com/docker/libcompose/vendor/github.com/Sirupsen/logrus
github.com/wunderkraut/radi-api/operation/command
github.com/wunderkraut/radi-api/operation/security
github.com/wunderkraut/radi-api/api
github.com/wunderkraut/radi-api/operation/config
github.com/wunderkraut/radi-api/builder
github.com/wunderkraut/radi-api/operation/setting
github.com/wunderkraut/radi-api/handler
github.com/wunderkraut/radi-handlers/configwrapper
github.com/wunderkraut/radi-api/operation/orchestrate
github.com/wunderkraut/radi-api/operation/project
github.com/wunderkraut/radi-handlers/bytesource
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/urlutil
github.com/docker/libcompose/vendor/github.com/docker/go-connections/nat
github.com/docker/libcompose/vendor/gopkg.in/yaml.v2
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/strslice
github.com/docker/libcompose/vendor/github.com/docker/go-units
github.com/docker/libcompose/vendor/github.com/flynn/go-shlex
github.com/docker/libcompose/yaml
github.com/docker/libcompose/vendor/github.com/xeipuuv/gojsonpointer
github.com/docker/libcompose/vendor/github.com/xeipuuv/gojsonreference
github.com/docker/libcompose/vendor/github.com/xeipuuv/gojsonschema
github.com/docker/libcompose/utils
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/blkiodev
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/mount
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/container
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/versions
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/filters
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/network
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/registry
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/swarm
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types
github.com/docker/libcompose/config
github.com/docker/libcompose/vendor/github.com/docker/docker/cliconfig/configfile
github.com/docker/libcompose/vendor/github.com/docker/distribution/digest
github.com/docker/libcompose/vendor/github.com/docker/distribution/registry/api/errcode
github.com/docker/libcompose/vendor/github.com/docker/distribution/uuid
github.com/docker/libcompose/vendor/github.com/gorilla/context
github.com/docker/libcompose/vendor/golang.org/x/net/context
github.com/docker/libcompose/vendor/github.com/gorilla/mux
github.com/docker/libcompose/vendor/github.com/docker/distribution/reference
github.com/docker/libcompose/vendor/github.com/docker/distribution/registry/client/transport
github.com/docker/libcompose/vendor/github.com/docker/docker/opts
github.com/docker/libcompose/vendor/github.com/docker/distribution/context
github.com/docker/libcompose/vendor/github.com/docker/distribution/registry/api/v2
github.com/docker/libcompose/vendor/github.com/docker/distribution
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/jsonlog
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/term
github.com/docker/libcompose/vendor/github.com/docker/distribution/registry/storage/cache
github.com/docker/libcompose/vendor/github.com/docker/distribution/registry/storage/cache/memory
github.com/docker/libcompose/vendor/github.com/docker/distribution/registry/client
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/jsonmessage
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/httputils
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/ioutils
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/random
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/stringid
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/tarsum
github.com/docker/libcompose/vendor/github.com/docker/distribution/registry/client/auth
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/fileutils
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/system
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/pools
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/promise
github.com/docker/libcompose/vendor/github.com/opencontainers/runc/libcontainer/system
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/idtools
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/mount
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/reexec
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/plugins/transport
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/archive
github.com/docker/libcompose/vendor/github.com/opencontainers/runc/libcontainer/user
github.com/docker/libcompose/vendor/golang.org/x/net/proxy
github.com/docker/libcompose/vendor/github.com/docker/go-connections/sockets
github.com/docker/libcompose/vendor/github.com/docker/go-connections/tlsconfig
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/chrootarchive
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/plugins
github.com/docker/libcompose/vendor/github.com/vbatts/tar-split/tar/storage
github.com/docker/libcompose/vendor/github.com/vbatts/tar-split/archive/tar
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/plugingetter
github.com/docker/libcompose/vendor/github.com/docker/docker/daemon/graphdriver
github.com/docker/libcompose/vendor/github.com/spf13/pflag
github.com/docker/libcompose/vendor/github.com/vbatts/tar-split/tar/asm
github.com/docker/libcompose/vendor/github.com/docker/docker/layer
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/homedir
github.com/docker/libcompose/vendor/github.com/docker/docker/cliconfig
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/events
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/reference
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/time
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/tlsconfig
github.com/docker/libcompose/vendor/github.com/pkg/errors
github.com/docker/libcompose/vendor/golang.org/x/net/context/ctxhttp
github.com/docker/libcompose/vendor/github.com/docker/docker/client
github.com/docker/libcompose/vendor/github.com/docker/docker/image
github.com/docker/libcompose/vendor/github.com/docker/docker/image/v1
github.com/docker/libcompose/vendor/github.com/docker/docker/reference
github.com/docker/libcompose/vendor/github.com/docker/docker/registry
github.com/docker/libcompose/logger
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/signal
github.com/docker/libcompose/vendor/github.com/docker/docker/runconfig/opts
github.com/docker/libcompose/docker/auth
github.com/docker/libcompose/project/events
github.com/docker/libcompose/project/options
github.com/docker/libcompose/version
github.com/docker/libcompose/vendor/golang.org/x/time/rate
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/progress
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/streamformatter
github.com/docker/libcompose/lookup
github.com/docker/libcompose/vendor/github.com/docker/docker/api/types/backend
github.com/docker/libcompose/vendor/github.com/docker/docker/builder/dockerignore
github.com/docker/libcompose/project
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/symlink
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/gitutils
github.com/docker/libcompose/vendor/github.com/docker/docker/builder
github.com/docker/libcompose/docker/client
github.com/docker/libcompose/docker/builder
github.com/docker/libcompose/vendor/github.com/docker/docker/pkg/stdcopy
github.com/docker/libcompose/docker/ctx
github.com/docker/libcompose/docker/network
github.com/docker/libcompose/labels
github.com/docker/libcompose/docker/container
github.com/docker/libcompose/docker/image
github.com/docker/libcompose/docker/volume
github.com/docker/libcompose/docker/service
github.com/wunderkraut/radi-api/operation/monitor
github.com/wunderkraut/radi-api/operation/document
github.com/wunderkraut/radi-handlers/null
github.com/rancher/go-rancher/client
github.com/docker/libcompose/docker
github.com/wunderkraut/radi-handlers/libcompose
github.com/wunderkraut/radi-handlers/local
github.com/Jalle19/upcloud-go-sdk/upcloud/request
github.com/Jalle19/upcloud-go-sdk/upcloud/service
github.com/wunderkraut/radi-api/operation/provision
github.com/wunderkraut/radi-handlers/upcloud
github.com/wunderkraut/radi-cli/version
github.com/wunderkraut/radi-handlers/rancher
github.com/wunderkraut/radi-cli/local
_/usr/src/myapp/main

 

Exited container

 **** Containerized build complete 

an executable binary has (hopefully) now been built 
in ./bin/radi


 **** Installation

This installer can now install the built binary for you,
if you don't want to do it manually.

The planned installation path is : /usr/local/bin

Would you like to me install a binary to that location? (y/n)

y
--> detected that sudo will be required, as you don't have write privelege to the target path
[sudo] password for james: 
"./build/make.sh" install
---> Make-bundle: install (in .)
  --> Installing last compiled binary (./bin/radi) to: /usr/local/bin

```

**Comment by james-nesbitt** [comment]
I enabled more verbose go build, because it takes a long time.

**Comment by james-nesbitt** [comment]
@aleksijohansson  please review again

https://github.com/wunderkraut/radi-cli/pull/14

**Comment by aleksijohansson** [comment]
Reviewed, accepted and merged #14

**Comment by james-nesbitt** [comment]
a quick note: we no longer call our wrapper build `build.sh` as the wundertools project already contains 2 scripts of that name.
